.github/workflows/build_and_run_test_app_usb.yml

```diff
@@ -0,0 +1,72 @@
+name: Build and Run USB Test Application
+
+on:
+  schedule:
+    - cron: '0 0 * * SAT' # Saturday midnight
+  pull_request:
+    types: [opened, reopened, synchronize]
+    paths:
+      - 'usb/**'
+      - 'workflows/build_and_run_test_app_usb.yml'
+
+jobs:
+  build:
+    name: Build USB Test App
+    strategy:
+      fail-fast: false
+      matrix:
+        idf_ver: ["release-v4.4", "release-v5.0", "release-v5.1", "release-v5.2", "latest"]
+        idf_target: ["esp32s2", "esp32s3"]
+    runs-on: ubuntu-20.04
+    container: espressif/idf:${{ matrix.idf_ver }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+      - name: Build USB Test Application
+        env:
+          IDF_TARGET: ${{ matrix.idf_target }}
+        shell: bash
+        working-directory: usb/test_app
+        run: |
+          . ${IDF_PATH}/export.sh
+          export PEDANTIC_FLAGS="-DIDF_CI_BUILD -Werror -Werror=deprecated-declarations -Werror=unused-variable -Werror=unused-but-set-variable -Werror=unused-function"
+          export EXTRA_CFLAGS="${PEDANTIC_FLAGS} -Wstrict-prototypes"
+          export EXTRA_CXXFLAGS="${PEDANTIC_FLAGS}"
+          idf.py build
```

Comment:
After we are sure that the CI infrastructure works, could you please refactor this using https://github.com/espressif/idf-build-apps ?

Example here: https://github.com/espressif/esp-bsp/blob/master/.github/workflows/build_example.yml#L43 (there is no need for parallel runs for now)